.github/workflows/pr-preview.yml

```
@@ -44,28 +31,102 @@ jobs:
+          name: main-app
+          path: dist
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: cosmos
+          path: cosmos-export
+
+      - name: Deploy main app to Surge
+        if: github.event.action != 'closed'
+        id: deploy-main
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2086:info:2:36: Double quote to prevent globbing and word splitting

(shellcheck)